### ISSUE TITLE
omrelp looses message

### ISSUE THREAD

**Issue by hasso** [rsyslog/rsyslog#247]
What I'm doing is simple "forward all logs to network host". Sender configuration:

```
*.* action(type="omrelp" target="172.16.108.130" port="10514"
           queue.spoolDirectory="/var/lib/rsyslog"
           queue.filename="forwarding" queue.type="LinkedList"
           queue.maxdiskspace="4g")
```
Receiver side:
```
template(name="RemoteClientLog" type="string"
         string="/var/log/remote/%HOSTNAME%-all.log")
ruleset(name="RemoteLogger") {
    action(type="omfile" dynaFile="RemoteClientLog")
}
input(type="imrelp" port="10514" ruleset="RemoteClient")
```

I'm generating log messages with logger and at the same time I'm restarting a receiver.
```
# for count in $(seq 1 200); do logger "Test $count."; sleep 0.1; done
```

What happens at receiver that one message gets lost:
```
2015-02-20T08:35:40.559862-08:00 loghost-1: user: Test 38.
2015-02-20T08:35:40.663667-08:00 loghost-1: user: Test 39.
2015-02-20T08:35:40.767253-08:00 loghost-1: user: Test 40.
2015-02-20T08:35:40.871081-08:00 loghost-1: user: Test 41.
2015-02-20T08:35:40.974780-08:00 loghost-1: user: Test 42.
2015-02-20T08:35:41.079088-08:00 loghost-1: rsyslogd-2007: action 'action 9' suspended, next retry is Fri Feb 20 08:36:11 2015 [try http://www.rsyslog.com/e/2007 ]
2015-02-20T08:35:41.181500-08:00 loghost-1: user: Test 44.
2015-02-20T08:35:41.284751-08:00 loghost-1: user: Test 45.
2015-02-20T08:35:41.387602-08:00 loghost-1: user: Test 46.
2015-02-20T08:35:41.490684-08:00 loghost-1: user: Test 47.
```

It's not reordering issue and the missing message with seq number 43 is correctly there in the messages file of the sender. It also happens with TCP transport (imtcp). Tested with 8.7.0 RPM packages for Centos 7 and 8.4.2.ad1 RPM packages for Centos 5.

**Comment by eseres** [comment]
I can confirm this critical issue, too. It is 100% repeatable.

**Comment by xuecai** [comment]
8.10.0 auso has this issus

**Comment by eseres** [comment]
With 8.12.0, I am still experiencing this.

**Comment by alorbach** [comment]
I think this is not an RELP only issue if @hasso is expiriencing the same issue with imtcp as well. 
It is proberly connected to this ISSUE https://github.com/rsyslog/rsyslog/issues/262

**Comment by ondrap** [comment]
I have this problem with RELP, tested with 7.4.4 on Ubuntu and 8.15 on Centos. I am sending it to a testing RELP server (not rsyslog), that starts returning 500 in the RELP protocol when some internal processing fails. The first message that generated the '500' error code is not resent by the rsyslog. When I kill the connection instead, rsyslog tends to resend the whole window (probably...), there is more message duplication, but it behaves better - however I cannot rule out that in some cases it wouldn't lose the message too.

**Comment by rgerhards** [comment]
this is fixed in 8.16.0
